website/docs/docs/building-a-dbt-project/metrics.md

```diff
@@ -90,6 +92,85 @@ metrics:
+| type        | The type of calculation to perform when evaluating a metric | count_distinct                  | yes       |
+| sql         | The expression to aggregate/calculate over                  | user_id                         | yes       |
+| timestamp   | The time-based component of the metric                      | signup_date                     | yes       |
+| time_grains | One or more "grains" at which the metric can be evaluated   | [day, week, month]              | yes       |
+| dimensions  | A list of dimensions to group or filter the metric by       | [plan, country]                 | no        |
+| filters     | A list of filters to apply before calculating the metric    | See below                       | no        |
+| meta        | Arbitrary key/value store                                   | {team: Finance}                 | no        |
+
+</VersionBlock>
+
+<VersionBlock firstVersion="1.2">
+
+### Available types
+
+| Metric Type    |  Description                                                               |
+|----------------|----------------------------------------------------------------------------|
+| count          | This metric type will apply the `count` aggregation to the specified field |
+| count_distinct | This metric type will apply the `count` aggregation to the specified field, with an additional distinct statement inside the aggregation |
+| sum            | This metric type will apply the `sum` aggregation to the specified field |
+| average        | This metric type will apply the `average` aggregation to the specified field |
+| min            | This metric type will apply the `min` aggregation to the specified field |
+| max            | This metric type will apply the `max` aggregation to the specified field |
+| expression     | This metric type is defined as any **non-aggregating** calculation of 1 or more metrics |
+
+</VersionBlock>
+
+<VersionBlock firstVersion="1.2">
+
+### Expression Metrics
+In v1.2, support was added for `expression` metrics, which are defined as non-aggregating calculations of 1 or more metrics. By defining these metrics, you are able to create metrics like:
+- ratios
+- subtractions 
+- any arbitrary calculation
+
+As long as the two+ base metrics (the metrics that comprise the `expression` metric) share the specified `time_grains` and `dimensions`, those attributes can be used in any downstream metrics macro.
+
+An example definition of an `expression` metric is:
+</VersionBlock>
+
+<VersionBlock firstVersion="1.2">
+
+```yaml
+# models/marts/product/schema.yml
+version: 2
+
+models:
+ - name: dim_customers
+   ...
+
+metrics:
+  - name: average_revenue_per_customer
+    label: Average Revenue Per Customer
+    description: "The average revenue received per customer"
+
+    type: expression
+    sql: "{{metric('total_revenue')}} / {{metric('count_of_customers')}}"
```

Comment:
[Non-blocking] Is this the syntax we'd recommend, over syntax like:
```yaml
    sql: "{{ metric('total_revenue') ~ "/" ~ metric('count_of_customers') }}"
```
(Does that alt syntax even work?)

Reply:
I tested the alt syntax and it didn't appear to work. Even testing a few different options around quotation placement returned unicode errors. 

To me that is fine anyway because I wouldn't want users to have to worry about concatenation within the sql field. Self contained makes most sense to me and personally feels the most ergonimic.